File: bookmarker-java/src/main/java/com/sivalabs/bookmarker/config/security/SecurityUser.java
package com.sivalabs.bookmarker.config.security;

import com.sivalabs.bookmarker.domain.entity.User;
import org.springframework.security.core.authority.SimpleGrantedAuthority;

import java.util.stream.Collectors;

public class SecurityUser extends org.springframework.security.core.userdetails.User {
    private User user;

    public SecurityUser(User user) {
        super(user.getEmail(), user.getPassword(),
                user.getRoles().stream()
                        .map(role -> new SimpleGrantedAuthority(role.getName())).collect(Collectors.toList()));
        this.user = user;
    }

    public User getUser() {
        return user;
    }
}

File: spring-boot-bookmarker-api/src/main/kotlin/com/sivalabs/bookmarker/security/SecurityUser.kt
package com.sivalabs.bookmarker.security

import org.springframework.security.core.authority.SimpleGrantedAuthority

class SecurityUser(val user: com.sivalabs.bookmarker.entity.User)
    : org.springframework.security.core.userdetails.User(
            user.email,
            user.password,
            user.roles.map { SimpleGrantedAuthority(it.name) }
        )
